sympy/vector/tests/test_parametricregion.py

```diff
@@ -0,0 +1,56 @@
+from sympy import sin, cos, pi, symbols
+from sympy.vector.coordsysrect import CoordSys3D
+from sympy.vector.parametricregion import ParametricRegion
+from sympy.testing.pytest import raises
+
+C = CoordSys3D('C')
+r, theta, phi, a, b, t = symbols('r theta phi a b t')
+
+def test_parametricregion():
+
+    point = ParametricRegion((), (3, 4), {})
+    assert point.definition == (3, 4)
+    assert point.parameters == ()
+    assert point.limits == {}
+
+    # line x = y
+    line_xy = ParametricRegion(C, (C.y), limits={C.y: (-3, 3)})
+    assert line_xy .definition == (C.y,)
+    assert line_xy.parameters == (C.x, C.y, C.z)
+
+    # line y = z
+    line_yz = ParametricRegion((t), (C.x,t,t), limits={t: (1, 2)})
+    assert line_yz.definition == (C.x,t,t)
+    assert line_yz.parameters == (t,)
+
+    p1 = ParametricRegion((a, b), (9*a, -16*b), limits={a: (0, 2), b: (-1, 5)})
+    assert p1.definition == (9*a, -16*b)
+    assert p1.parameters == (a, b)
+    assert p1.limits == {a: (0, 2), b: (-1, 5)}
+
+    p2 = ParametricRegion(t, (t, t**3))
+    assert p2.parameters == (t,)
+    assert p2.limits == {}
+
+    circle = ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), {theta: (0, 2*pi)})
+    assert circle.definition == (r*cos(theta), r*sin(theta))
+
+    halfdisc = ParametricRegion((r, theta), (r*cos(theta), r* sin(theta)), {r: (-2, 2), theta: (0, pi)})
+    assert halfdisc.definition == (r*cos(theta), r*sin(theta))
+    assert halfdisc.parameters == (r, theta)
+    assert halfdisc.limits == {r: (-2, 2), theta: (0, pi)}
+
+    ellipse = ParametricRegion(t, (a*cos(t), b*sin(t)), {t: (0, 8)})
+    assert ellipse.parameters == (t,)
+    assert ellipse.limits == {t: (0, 8)}
+
+    cylinder = ParametricRegion((r, theta, C.z), (cos(theta), r*sin(theta), C.z), {theta: (0, 2*pi), C.z: (0, 4)})
+    assert cylinder.parameters == (r, theta, C.z)
```

Comment:
I see you have introduced `parameters_or_coordsys` only because you have to deal with coordinate systems `C`, but as you mix coordinate scalars and ordinary symbols in this way, you can always specify the region regardless of the ambient space of the coordinate systems and it loses no generality to use ordinary symbols only (`x, y, z` instead of `C.x, C.y, C.z`).
And also what would be the necessary reason to pass `C` in place of `(C.x, C.y, C.z)`

Reply:
> And also what would be the necessary reason to pass `C` in place of `(C.x, C.y, C.z)` if they are implicitly converted any way?

Shortcut?

Reply:
> it loses no generality to use ordinary symbols like x, y, z everywhere in place of coordinate systems.

Yes, I think we can replace base scalars with any variable. Suppose we want to represent a rectangle:
```python
ParametricRegion(C, (C.x, C.y), {C.x: (1, 2), C.y: (3, 5)})
```
We can write this as
```python
ParametricRegion((x, y), (x, y), {x: (1, 2), y: (3, 5)})
# C.x = x,  C.y = y, C.z = C.zd
```

Reply:
Just to make this explicit,
When we define the definition tuple ..suppose `(r*cos(theta), r*(sin(theta), r*sin(phi))`, this implies
```
C.x = r*cos(theta), C.y = r*(sin(theta), C.y = r*sin(phi)
```
These base scalars are assumed. They may be determined from the base scalars of the vector/scalar field for integration.
The only reason I think we may need a coordinate system variable other than for shortcut is when one needs to specify base scalars (instead of using base scalars of the vector/scalar field).  

```
p = ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), limits={r: (0,1), theta: (0, pi)})
Integral(C.x**2*C.y, p)
# C.x = r*cos(theta),  C.y = r*sin(theta), C.z = C.z
Integral(R.x*R.y*R.z, p)
# R.x = r*cos(theta),  R.y = r*sin(theta), R.z = R.z
```

Reply:
What happens if it is `Integral(R.x*C.y, p)`?
Perhaps, `p` can have its own base scalars?

Reply:
Yes, I am in favor of this. But there should be some relationship between the coordinate system of vector/scalar field and the coordinate system of the parametric region. For calculating the integral, we need to replace base scalars in the field with parameters. This requires some connection between them.

@Upabjojr What do you think?

Reply:
My original thought was just to use parameters. Mixing parameters and base scalars in the tests is a bit strange, but it's OK with me.

Reply:
I think for the tests, we can leave them as it is. But for users, it should be preferred to not mix them.